src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -39,6 +39,11 @@ class FrameworkExtension extends Extension
     private $sessionConfigEnabled = false;
 
     /**
+     * @var string|null
+     */
+    private $kernelRootHash;
```

Comment:
In fact it's not true to name it `kernelId` because it's the hash of the root directory location, not of the kernel (several kernels can be in the same directory and will share the same hash). In the current use it makes sense because several kernels of the same app can share things in the cache (like validator and serializer metadata).